manager/src/main/java/com/redhat/service/smartevents/manager/connectors/ConnectorsApiClientImpl.java

```diff
@@ -67,15 +67,19 @@ public Connector getConnector(String connectorExternalId) {
         ConnectorsApi connectorsAPI = createConnectorsAPI();
 
         try {
-            if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug(String.format("Retrieving Connector with ID '%s'", connectorExternalId));
-            }
+            LOGGER.debug("Retrieving Connector with ID '{}'", connectorExternalId);
             return connectorsAPI.getConnector(connectorExternalId);
         } catch (ApiException e) {
-            if (e.getCode() == HttpStatus.SC_NOT_FOUND) {
-                return null;
+            switch (e.getCode()) {
+                case HttpStatus.SC_NOT_FOUND:
+                case HttpStatus.SC_GONE:
```

Comment:
Could you please add a clarifying comment on why these two branches are handled differently? 

Something like ```// this is expected, don't do anything ``` and ``` // unknown status code```